docs/content/testing/testing-guidelines.md

```diff
@@ -0,0 +1,265 @@
+**Warning:** The [pre-alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) release is for evaluation purposes only, with limited documentation - see the guidance on [Recommended use](../../../README.md#recommended-use).
+
+-----
+
+# Testing Guidelines
+
+The tests should:
+
+- Set up some initial state
+- Execute a method that acts upon this state and results in a new state
+- Make assertions in the new state against the expected changes
+- Clean up
+
+NUnit allows this to be done through the use of Fixtures and special method annotations.
+
+## Module Directory Structure
+
+The Unity GDK project's assets are divided into modules, such as `Core`, `Physics`, `PlayerLifecycle`.
+
+Each module is in its own directory within `workers\unity\Assets\Gdk`.
+
+Each module has its own tests that cover the functionality of that module.
+
+Unity Test Runner also allows separate playmode and editmode tests. For more information on this division, please refer to the [Unity Test Runner manual page](https://docs.unity3d.com/Manual/testing-editortestsrunner.html).
+
+Each module folder will contain the tests about that module, and within separate subdirectories if a module needs both editmode and playmode testing - this is a requirement from Unity:
+
+> The Editmode and Playmode folders need to be separate as the tests need to be in different assemblies, and currently Unity `.asmdef` files affect all files within the same directory.
+
+The folder structure for a module should look like this:
+- `<ModuleName>` (e.g. `Core`)/
+  - Improbable.Gdk.`<ModuleName>`.asmdef
+  - `Some/Folder/<ClassName>`.cs
+  - `Some/Other/Folder/<ClassName>`.cs
+  - Tests/
+    - Editmode/
+      - Improbable.Gdk.`<ModuleName>`.EditmodeTests.asmdef
+      - `Some/Folder/<ClassName>`Tests.cs
+    - Playmode/
+      - Improbable.Gdk.`<ModuleName>`.PlaymodeTests.asmdef
+      - `Some/Other/Folder/<ClassName>`Tests.cs
+
+Test assemblies must match this pattern (see [Namespaces](#namespaces)):
+> `Improbable.Gdk.<ModuleName>.<TestMode>Tests.asmdef`
+
+If a class to be tested will have both Editmode and Playmode tests, you can name the test filenames as `<ClassName>EditmodeTests.cs` and `<ClassName>PlaymodeTests.cs` within their respective assemblies.
+
+Within each test folder, the folder structure should match the non-test code structure.
+
+If any file path can be broken down into this pattern:
+
+> `<module path>`/`<path within module>`.cs
+
+The Editmode test for this class should be in:
+
+> `<module path>`/Tests/Editmode/`<path within module>`Tests.cs
+
+For a more concrete example, if you look at this class:
+
+workers/unity/Assets/Gdk/`Core`/`Utility/CommandLineUtility`.cs
```

Comment:
highlight the whole path:
`workers/unity/Assets/Gdk/Core/Utility/CommandLineUtility.cs`